kano_settings/system/proxy.py

```diff
@@ -50,6 +54,63 @@ def set_chromium(enable, host=None, port=None):
     return
 
 
+def set_midori(enable, host=None, port=None, user_agent='Mozilla/5.0 (X11; Linux) AppleWebKit/535.22+ Midori/0.4'):
+
+    # Read Midori configuration file in
+    with open(midori_cfg, 'r') as f:
+        config_lines=f.readlines()
+
+    def conf_add_or_replace(item, value, config_lines):
+        '''
+        A function to simplify editing the configuration file values
+        add the item if not found, replace its value otherwise
+        '''
+        found=False
+        newitem = '{}={}\n'.format(item, value)
+        for idx, line in enumerate(config_lines):
+            if line.startswith(item):
+                config_lines[idx]=newitem
+                found=True
+
+        if not found:
+            config_lines.append(newitem)
+
+        return newitem
```

Comment:
This function could perhaps be made easier (if you don't care about the order of config file) by just doing:

```python
def conf_add_or_replace(item, value, config_lines):
    '''
    A function to simplify editing the configuration file values
    add the item if not found, replace its value otherwise
    '''
    conf_remove(item, config_lines)

    newitem = '{}={}\n'.format(item, value)
    config_lines.append(newitem)

    return newitem
```

But if you do care about config file order then this isn't viable.